packages/calypso-e2e/src/lib/components/editor-settings-sidebar-component.ts

```diff
@@ -251,28 +352,30 @@ export class EditorSettingsSidebarComponent {
 	 * @param {string} slug URL slug to set.
 	 */
 	async enterUrlSlug( slug: string ) {
-		await this.frame.fill( selectors.urlSlugInput, slug );
-		// Playwright is so fast, let's make sure the post's/page's state actually updates in case we're going to publish or update right after.
-		// By adding the forward slash, we ensure the new route has been appended to the fully qualified URL in the sidebar.
-		await this.frame.waitForSelector( `text=/${ slug }` );
+		const inputLocator = this.frameLocator.locator( selectors.permalinkInput );
+		await inputLocator.fill( slug );
+		// Hit the Tab key to confirm URL slug input and update the Post URL
+		// shown in this section.
+		await this.page.keyboard.press( 'Tab' );
+
+		const generatedURLSlugLocator = this.frameLocator.locator( selectors.permalinkGeneratedURL );
+		const generatedSlug = await generatedURLSlugLocator.innerText();
+		const expectedSlug = slug.replace( ' ', '-' );
+
+		if ( ! generatedSlug.includes( expectedSlug ) ) {
+			throw new Error(
+				`Generated URL slug ${ generatedSlug } did not match expected URL slug ${ expectedSlug }`
+			);
+		}
```

Comment:
Same thing re: race conditions -- any reason to just not wait for the expected new URL? It also seems like it might be a bit simpler.

Reply:
Even in that case, we might want to narrow the selector somewhat.

Reply:
Used a bit more narrowed but simpler method of checking and it seems to work 👍